src/router/mod.rs

```diff
@@ -433,12 +424,15 @@ mod test {
         assert_eq!(router.current_history_index(), 2);
     }
 
+    // Testing return value and side effects of Router::back
```

Comment:
This commit includes a fair bit of addition to tests, but I chose to include it, as there are no new tests, only an update for existing tests to be in line with changes made, and some extra asserts within them that I thought might be a nice little extra.